Converting tests

prury opened the pull request:
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

Some new automated test coverage, will be improved in the future. 
reason why i did not use for-each: 
![image](https://github.com/kodadot/nft-gallery/assets/36627808/7126a4b9-ed36-4d08-83fd-e02d02ca31f9)

MerchShop was failing when waiting for page to load, so i took it off for now

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [ ] Related to: #7051 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Community participation

- [ ] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

## Copilot Summary
copilot:summary

copilot:poem

roiLeo commented on components/TheFooter.vue:
```diff
@@ -1,13 +1,13 @@
 <template>
-  <footer class="footer-container section">
+  <footer class="footer-container section" data-testid="entire-footer">
```
I would name it `footer-container` we need to unify test-id with better naming I'm thinking like css we can use Blocks – Elements – Modifiers

prury replied:
allright!

roiLeo commented on tests/e2e/createcollection.spec.ts:
```diff
@@ -0,0 +1,60 @@
+    // defines a variable containing the collection name data-testid
+    const collectionName = page.getByTestId('collection-name')
+    // Fill and check if Collection Name works properly
+    await collectionName.locator('input').fill('Kodadot Automatic Collection')
+    await collectionName.locator('input').fill('')
```
2 times filled?

prury replied:
filled with text to see if the field works and then filled with nothing and clicked elsewhere in order to trigger warning on field, should i do it in another way?

roiLeo replied:
I would create separete test for these thing something like 
```
test('Show warning message on field when collection name empty, {....})
```

prury replied:
hmmmm, ok!

roiLeo commented on tests/e2e/footer.spec.ts:
```diff
@@ -0,0 +1,74 @@
+    await newTab.waitForLoadState()
+    await expect(newTab).toHaveURL(footerLinks[i].linkAddress)
+    await newTab.close()
+    i++
```
is `i` needed?

prury replied:
is there a way to access current link adress without it?

roiLeo replied:
oops nevermind didn't saw `footerLinks[i].linkAddress` lol

roiLeo reviewed (commented):
I wouldn't trust `.o-icon` or `.o-input` selector maybe add test-id

**before**
![Screenshot 2023-09-14 at 09-04-49 fix Teleport Bridge Rename Statemine to AssetHub Kusama · kodadot_nft-gallery@3c28fd6](https://github.com/kodadot/nft-gallery/assets/9987732/492f045b-7442-4c5e-bbb3-9514d73f073e)

**after**
![Screenshot 2023-09-14 at 09-05-06 New Playwright tests and modifications · kodadot_nft-gallery@de69f56](https://github.com/kodadot/nft-gallery/assets/9987732/46e44f3e-2dc2-49a5-9b33-071d33c5faba)

+1min for not so many components, can you try to reduce it?

prury commented:
> I wouldn't trust `.o-icon` or `.o-input` selector maybe add test-id
> 
> **before** ![Screenshot 2023-09-14 at 09-04-49 fix Teleport Bridge Rename Statemine to AssetHub Kusama · kodadot_nft-gallery@3c28fd6](https://user-images.githubusercontent.com/9987732/267887560-492f045b-7442-4c5e-bbb3-9514d73f073e.png)
> 
> **after** ![Screenshot 2023-09-14 at 09-05-06 New Playwright tests and modifications · kodadot_nft-gallery@de69f56](https://user-images.githubusercontent.com/9987732/267887576-46e44f3e-2dc2-49a5-9b33-071d33c5faba.png)
> 
> +1min for not so many components, can you try to reduce it?

will make the changes, ty. 
i can also try to activate parallelism, since it seems that we are only using one worker for tests

prury commented on components/create/CreateCollection.vue:
```diff
@@ -26,6 +26,7 @@
         :label="`${$t('mint.collection.name.label')} *`"
         required
+        data-testid="collection-name"
         :error="!name">
```
@roiLeo where to insert a data-testid specific for the warning?

roiLeo replied:
You don't need to, just check if `o-field__message-danger` is present neer this input

Btw it's a test that is not too relevant since it's html that will check if input is present or not & you won't be able to submit form

roiLeo replied:
idk who wrote the code but `required` prop on `NeoField` is useless and don't work

prury replied:
when i wrote i thought about a case when dangers messages were not being displayed. Will leave only `o-field__message-danger` ty!

prury commented:
reduced by 30s, lmk if that works

roiLeo commented on tests/e2e/createcollection.spec.ts:
```diff
@@ -0,0 +1,67 @@
+  //active unlimited items switch
+  const collectionAmount = page.getByTestId('collection-maxAmount')
+  await collectionAmount.locator('.o-switch').click()
+  await collectionAmount.locator('.o-switch').click()
```
two times click?

prury replied:
ty, removed the second one and added a line to check if the related input shows up

roiLeo commented on tests/e2e/redirect.spec.ts:
```diff
@@ -11,6 +11,7 @@ test('Should redirect from rmrk2 prefix to ksm', async ({ page }) => {
 
   await page.goto('/rmrk2/explore/items?listed=true&sort=updatedAt_DESC')
+  await page.waitForLoadState()
   expect(await page.url()).toContain(
```
await should be before expect

prury replied:
changed the syntax

prury commented:
seems that the playwright actions its taking more than 10m and it stops itself automatically(building + installing browsers + tests)
should i improve action time? 
improve test code? 

i appreciate any help @kodadot/internal-dev 

using 1 worker locally: 
![image](https://github.com/kodadot/nft-gallery/assets/36627808/a9eff19d-f054-4d13-9211-507e0802f3b9)

using 6 workers locally: 
![image](https://github.com/kodadot/nft-gallery/assets/36627808/dc3cc9a4-4556-4df2-8565-8c4a606f61f3)

daiagi commented on tests/e2e/footer.spec.ts:
```diff
@@ -0,0 +1,126 @@
+test('Check Footer links', async ({ page }) => {
+  let i = 0
+  await page.goto('/')
+  for (const data of footerLinks) {
```
code sonar complain about 3 code smells related to these loops

`data` is not used

prury replied:
any tips on how can i solve that?

daiagi replied:
```ts
for (const [index, _] of footerLinks.entries())
```

honestly, why have we disallowed regular for loops in eslint?

prury commented:
@daiagi pointed that the app is building twice, will try to reuse same workflow to run tests

prury commented:
Report: 
- tried to run playwright tests on the same build it uses for unity tests but was unable to
- reverted everything back to what it were before
- realized tests were not passing anymore after the reverted changes(they were passing successful before trying step 1)
- realized that in the middle of the process the bot updated playwright package, causing some commands not to run anymore
- simplified the test that was not passing in order to fit new package
